TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
 ]
 
-{ #category : #'tests comparing' }
-TPRBParserTest >> testEquivalentExceptRenaming [
-	#(#('a 3-4' 'a 4-3' false) #('a #[3 4]' 'a #(3 4)' false) #('a variable1 ~~ "comment" variable2' 'a variable1 ~~ variable2' true) #('a variable1' 'a variable2' false) #('a [:a :b | a + b]' 'a [:b :a | a + b]' false) #('a | a b | a + b' 'a | b a | a + b' true) #('a | a | a msg1; msg2' 'a | b | b msg2; msg2' false) #('a c' 'a d' true) #('a | a b | a := b. ^b msg1' 'a | a b | b := a. ^a msg1' true) #('a | a b | a := b. ^b msg1: a' 'a | a b | b := a. ^b msg1: a' false) #('a: b b + 4' 'a: e e + 4' true) #('a: b b + 4' 'b: b b + 4' false) #('a: b b: c b + c' 'a: c b: b c + b' true) #('a: a b: b a + b' 'a: b b: a a + b' false))
-		do: [ :each | 
-			self
-				assert: ((self parserClass parseMethod: each first) equalTo: (self parserClass parseMethod: (each at: 2)) exceptForVariables: #('c'))
-				identicalTo: each last ]
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageBadlyParenthesizedEndOfStatement [
+	"Should propose a closing parenthese before the end of the statement. Should also offer possibility to quickly erase it."
+	self should: [self parserClass parseExpression: '(temp := 1 + 3 .'] raise: SyntaxErrorNotification.
 ]
 
-{ #category : #'tests parsing' }
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageEncouteringEndOfStatement [
+	"Could propose multiple reparations like assignement, period, array..."
+	self should: [self parserClass parseExpression: 'temp 1 + 3'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageFaultyCascadeGivesMultipleErrorAndNoMassage [
+	| tree |
+	tree := self parseFaultyExpression: 'faulty cascade; 1; node'.
+	self assert: tree isSequence.
+	self assert: tree statements size equals: 2.
+	
+	"The first node is meant to be a cascade but the failure prevent from having a cascade node which means having an error 
+	 node instead. This has some repercutions : their is no cascade message and the name of the parents are wrong."
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.
+  1; node'.
+	
+	"The problem is further enlightenned with longer cascades."
+	tree := self parseFaultyExpression: 'faulty cascade; successful; until; not; 1; node'.
+	self assert: tree isSequence.
+	self assert: tree statements size equals: 2.
+	
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.

```

Comment:
this one is a cascade without receiver

Reply:
MAybe should be in a different test

Reply:
This one is the printString of the method node with the recomposed newSource.